dotenv/main.py

```diff
@@ -181,21 +181,42 @@ def unset_key(dotenv_path, key_to_unset, quote_mode="always"):
 
 
 def resolve_nested_variables(values):
-    def _replacement(name):
+    def _replacement(name, default=None):
         """
         get appropriate value for a variable name.
         first search in environ, if not found,
         then look into the dotenv variables
         """
-        ret = os.getenv(name, new_values.get(name, ""))
-        return ret
+
+        # If name is set as an environment variable, use its value
+        if name in os.environ:
+            return os.getenv(name)
+
+        # If name is the same as k, we need to handle the situation special, or we will have ${K}
+        # or ${K:-default} as the return value.  Note that k comes from the for loop below
+        if name == k:
+            # If we have a default set, return it
+            if default:
```

Comment:
How about `return default or ""`?

Reply:
That was the original line of code; i don’t know why i changed it, but i will put that back in place.